File: LeetCode-Programs/Divide a String Into Groups of Size k.py
class Solution:
    def divideString(self, s: str, k: int, fill: str) -> List[str]:
        new_list=[]
        strr=""
        for i in range(0, len(s), k):
            strr=s[i:i+k]
            if len(strr)==k:
                new_list.append(strr)
            else:
                while len(strr)!=k:
                    strr+=fill
                new_list.append(strr)
        return new_list

File: LeetCode-Programs/Divide a String Into Groups of Size k.cpp
class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        vector<string> vec;
        string str;
        
        for(int i=0;i<s.size();i+=k){
            str=s.substr(i, k);
            if(str.size()==k)
                vec.push_back(str);
            else{
                while(str.length()!=k)
                    str+=fill;
                vec.push_back(str);
            }
        }
        return vec;
    }
};
